File: app/src/main/java/mishka/rssreader/ui/BaseActivity.java
package mishka.rssreader.ui;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import mishka.rssreader.RssApplication;
import mishka.rssreader.di.component.ApplicationComponent;

public abstract class BaseActivity extends AppCompatActivity {
    protected ApplicationComponent getApplicationComponent(){
        return ((RssApplication)getApplication()).getApplicationComponent();
    }
}

File: app/src/main/java/mishka/rssreader/ui/BaseActivity.kt
package mishka.rssreader.ui;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import mishka.rssreader.RssApplication;
import mishka.rssreader.di.component.ApplicationComponent;

abstract class BaseActivity : AppCompatActivity() {
    public fun getApplicationComponent() = (application as RssApplication).applicationComponent
}
